### PR TITLE
executor docs refresh

### PR DESCRIPTION
**Pull request by gilescope**
No code changes.

Removed duplication between module docs and readme.md. Apparently this is the 'right' way to do it these days.

Please dive in with any suggestions on how we can re-word the docs to make anything clearer.
(And also if I've misunderstood anything and am twisting the original docs meaning)

### CODE REVIEW

**Review by expenses — APPROVED** [review]
I think this looks good, I haven't checked over the executor code to see if it matches though.

File: client/executor/wasmtime/src/runtime.rs [hunk 1]
```diff
@@ -279,7 +279,7 @@ fn common_config(semantics: &Semantics) -> std::result::Result<wasmtime::Config,
 /// usage in bytes.
 ///
 /// The actual number of bytes consumed by a function is not trivial to compute  without going through
-/// full compilation. Therefore, it's expected that `native_stack_max` is grealy overestimated and
+/// full compilation. Therefore, it's expected that `native_stack_max` is greedily overestimated and
```
Comment (expenses) [inline client/executor/wasmtime/src/runtime.rs:282]: Pretty sure it's meant to be:

```suggestion
/// full compilation. Therefore, it's expected that `native_stack_max` is greatly overestimated and
```

**Comment by gilescope** [comment]
Great build error:
`error[E0658]: arbitrary expressions in key-value attributes are unstable`
Except they're not:
https://github.com/rust-lang/rust/pull/83366
Hmm and that was a while ago....

File: client/executor/common/src/lib.rs [hunk 2]
```diff
@@ -16,8 +16,7 @@
 // You should have received a copy of the GNU General Public License
 // along with this program. If not, see <https://www.gnu.org/licenses/>.
 
-//! A set of common definitions that are needed for defining execution engines.
-
+#![doc = include_str!("../README.md")]
```
Comment (bkchr) [inline client/executor/common/src/lib.rs:19]: Can you please move the docs back here.

**Review by bkchr — CHANGES_REQUESTED** [review]

File: client/executor/common/src/lib.rs (hunk 2 above)
Comment (gilescope) [reply client/executor/common/src/lib.rs:19]: Ok, I can see I'm too early somehow for this change. Will do.

Comment (bkchr) [reply client/executor/common/src/lib.rs:19]: We also don't use this anywhere and I'm not sure that stuff like intra doc links would still work. I honestly would not loose such a good feature for having an external file that some editor can understand better.

**Pull request closed without merging**